packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/tile_overlay.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'types.dart';
+import 'package:meta/meta.dart' show immutable, required;
+
+/// Uniquely identifies a [TileOverlay] among [GoogleMap] tile overlays.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class TileOverlayId {
+  /// Creates an immutable identifier for a [TileOverlay].
+  TileOverlayId(this.value) : assert(value != null);
+
+  /// value of the [TileOverlayId].
+  final String value;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final TileOverlayId typedOther = other;
```

Comment:
Please use the style from here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode

Reply:
done